Apps/AdminWebClient/src/ClientApp/src/views/Communication.vue

```diff
@@ -86,13 +109,172 @@ export default class CommunicationView extends Vue {
     message: ""
   };
 
+  private tableHeaders: any[] = [
+    {
+      text: "Subject",
+      value: "subject",
+      width: "30%"
+    },
+    {
+      text: "Effective On",
+      value: "effectiveDateTime",
+      width: "20%"
+    },
+    {
+      text: "Expires On",
+      value: "expiryDateTime",
+      width: "20%"
+    },
+    {
+      text: "Text",
+      value: "text",
+      width: "40%"
+    }
+  ];
+
+  private communicationList: Communication[] = [];
+
   private communicationService!: ICommunicationService;
 
   mounted() {
     this.communicationService = container.get(
       SERVICE_IDENTIFIER.CommunicationService
     );
     this.clearForm();
+    this.loadCommunicationList();
+  }
+
+  private sortCommunicationListByEffectiveDate(isDescending: boolean) {
+    this.communicationList.sort((a, b) => {
+      if (a.effectiveDateTime > b.effectiveDateTime) {
+        return isDescending ? -1 : 1;
+      } else if (a.effectiveDateTime < b.effectiveDateTime) {
+        return isDescending ? 1 : -1;
+      }
+      return 0;
+    });
+  }
+
+  private sortCommunicationListByExpiryDate(isDescending: boolean) {
+    this.communicationList.sort((a, b) => {
+      if (a.expiryDateTime > b.expiryDateTime) {
+        return isDescending ? -1 : 1;
+      } else if (a.expiryDateTime < b.expiryDateTime) {
+        return isDescending ? 1 : -1;
+      }
+      return 0;
+    });
+  }
+
+  private customSort(
+    items: Communication[],
+    index: any[],
+    isDescending: boolean[]
+  ) {
+    // items: 'Communication' items
+    // index: Enabled sort headers value. (black arrow status).
+    // isDescending: Whether enabled sort headers is desc
+    if (index === undefined) {
+      index = ["effectiveDateTime"];
+      isDescending = [true];
+    }
+
+    if (index[0] === "effectiveDateTime") {
+      this.sortCommunicationListByEffectiveDate(isDescending[0]);
+    } else if (index[0] === "expiryDateTime") {
+      this.sortCommunicationListByExpiryDate(isDescending[0]);
+    }
+    return this.communicationList;
```

Comment:
Will the customSort method be called for all sorting? if that is the case sorting by the other columns will not work

Reply:
correct, per requirements on the story #8209, we only need sorting feature for these 2 dates.

Reply:
Ah, interesting. I am not sure if that means "only sorteable by dates". I read as a "Date columns must be sorted". Either way we can create a bug and talk it later.

Reply:
oh I see :-) as this is the new list to be implemented. We may need to add an acceptance criteria to say "sotable by all the columns including dates". I leave it as is for now then even it is an easy change.